### PR TITLE
review and fix Amplify iOS Storage

### PR DESCRIPTION
**Pull request by drochetti**
**Notes:**

- moved content to the right pages
- reformatted code snippets
- reviewed faulty snippets

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/storage/fragments/ios/configureaccess.md [hunk 1]
```diff
@@ -16,7 +17,3 @@ Another user that wants to read the file can specify the user that created it:
 let options = StorageDownloadDataRequest.Options(accessLevel: .protected,
                                                  targetIdentityId: "OtherUserId")
 ```
-
-## Private Access
```
Comment (brene) [inline docs/lib/storage/fragments/ios/configureaccess.md:20]: are we removing these sections altogether?

Comment (lawmicha) [reply docs/lib/storage/fragments/ios/configureaccess.md:20]: looks like the sections were empty? under this "configure file access levels" section, i think it's important to have them like. 
here's the JS equivalant in the old docs https://aws-amplify.github.io/docs/js/storage#file-access-levels
In the new JS docs: https://docs.amplify.aws/lib/storage/configureaccess?platform=js

we should just use the same for consistency, even though i recall in one of the naming meetings, we decided to rename "public" to "guest".

File: docs/lib/storage/fragments/ios/getting-started.md [hunk 2]
```diff
@@ -61,22 +57,25 @@ If this is a new project, run `pod init` from the root of your application folde
 ```ruby
 target :'YOUR-APP-NAME' do
 use_frameworks!
-    pod 'AmplifyPlugins/AWSS3StoragePlugin'
-    pod 'AWSMobileClient', '~> 2.13.0'
+  pod 'AmplifyPlugins/AWSS3StoragePlugin'
+  pod 'AWSMobileClient', '~> 2.13.0'
```
Comment (lawmicha) [inline docs/lib/storage/fragments/ios/getting-started.md:61]: i think the new docs have removed all versions for the podFile, ie in Analytics
```
target :'YOUR-APP-NAME' do
  use_frameworks!
  pod 'AmplifyPlugins/AWSPinpointAnalyticsPlugin'
  pod 'AWSMobileClient'
end
```

File: docs/lib/storage/fragments/ios/getting-started.md [hunk 3]
```diff
@@ -89,41 +88,44 @@ open <YOURAPP>.xcworkspace
...
+    AWSMobileClient.default().initialize { (userState, error) in
+        guard error == nil else {
+            print("Error initializing AWSMobileClient. Error: \(error!.localizedDescription)")
+            return
         }
+        print("AWSMobileClient initialized, userstate: \(userState)")
+    }
 
-        return true
+    let storagePlugin = AWSS3StoragePlugin()
```
Comment (lawmicha) [inline docs/lib/storage/fragments/ios/getting-started.md:113]: can move this to `try Amplify.add(plugin: AWSS3StoragePlugin()`

File: docs/lib/storage/fragments/ios/configureaccess.md (hunk 1 above)
Comment (drochetti) [reply docs/lib/storage/fragments/ios/configureaccess.md:20]: We should have them indeed but it's out of the scope of this PR. Those sections are non-existent in the [current docs](https://aws-amplify.github.io/docs/ios/storage#restrict-access) and the goal is to bring them to parity (as long as it's correct).

One of the goals for Amplify iOS/Android GA is to review the documentation content, so we can address the missing content as part of that work.

Comment (brene) [reply docs/lib/storage/fragments/ios/configureaccess.md:20]: Opened an issue https://github.com/aws-amplify/docs/issues/1586 to not lose track.

File: docs/lib/storage/fragments/ios/getting-started.md (hunk 3 above)
Comment (drochetti) [reply docs/lib/storage/fragments/ios/getting-started.md]: done!

File: docs/lib/storage/fragments/ios/getting-started.md (hunk 2 above)
Comment (drochetti) [reply docs/lib/storage/fragments/ios/getting-started.md]: done!

**Pull request merged**